### PR TITLE
[SYCL][Graph] Add graph query function ...

### PR DESCRIPTION
**Pull request by reble**
... for a recording queue.

One intended use-case is pausing the recording frame and eagerly submit work, like in the following:

```cpp
void foo(sycl::queue my_queue, /* ... */) {
  if(my_queue.ext_oneapi_get_state() == experimental::queue_state::recording) {
    auto my_graph = my_queue.ext_oneapi_get_graph();
    my_graph.end_recording();
    // submit work eagerly here, outside the recording frame
    my_queue.begin_recording(my_graph);
  }
}
```

Another option would be adding an `end_recording` function to a queue that returns a `command_graph`.

### CODE REVIEW

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc [hunk 1]
```diff
@@ -1019,6 +1022,22 @@ Returns: If the queue is in the default state where commands are scheduled
...
+|
+[source,c++]
+----
+command_graph<graph_state::modifiable>
+queue::ext_oneapi_get_graph();
+----
+
+| Query the underlying command graph of a queue when recording.
+
+Returns: A copy of the graph object, the queue is recording commands into.
```
Comment (EwanC) [inline sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc]:
```suggestion
Returns: The graph object that the queue is recording commands into.
```
I'd remove the word "copy" from this sentence. The command_graph objects are reference counted so we don't need this language, and "copy" might make users think that if they modify the returned command_graph object in some way, e.g. add a node, that wouldn't affect the original.

**Review by EwanC — COMMENTED** [review]
Proposal makes sense to me 👍

File: sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc [hunk 2]
```diff
@@ -1019,6 +1022,22 @@ Returns: If the queue is in the default state where commands are scheduled
 immediately for execution, `queue_state::executing` is returned. Otherwise,
 `queue_state::recording` is returned where commands are redirected to a `command_graph`
 object.
+|
+[source,c++]
+----
+command_graph<graph_state::modifiable>
+queue::ext_oneapi_get_graph();
```
Comment (reble) [inline sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc]:
```suggestion
queue::ext_oneapi_get_graph() const;
```

**Comment by reble** [comment]
Moved to an upstream PR https://github.com/intel/llvm/pull/12460

**Pull request closed without merging**